### PR TITLE
Code improvement from black box branch merge to master

### PR DESCRIPTION
**Pull request by shokakucarrier**

### CODE REVIEW

File: src/main/java/org/commonjava/util/sidecar/interceptor/MetricsHandlerInterceptor.java [hunk 1]
```diff
@@ -35,6 +36,7 @@
 
 @Interceptor
 @MetricsHandler
+@RegisterForReflection
```
Comment (ligangty) [inline src/main/java/org/commonjava/util/sidecar/interceptor/MetricsHandlerInterceptor.java:39]: Why should this Interceptor use this RegisterForReflection annotation here?

**Review by ligangty — COMMENTED** [review]
Generally LGTM except a question for reflection.

File: src/main/java/org/commonjava/util/sidecar/interceptor/MetricsHandlerInterceptor.java (hunk 1 above)
Comment (shokakucarrier) [reply src/main/java/org/commonjava/util/sidecar/interceptor/MetricsHandlerInterceptor.java:39]: It's for native compilation, do we want to remove everything related to that?

Comment (ligangty) [reply src/main/java/org/commonjava/util/sidecar/interceptor/MetricsHandlerInterceptor.java:39]: Yes I know it's for native compilation. Question here is: is any reflection code used in this interceptor? Or is is a must for Quarkus framework to register all CDI interceptors with this annotation? If yes of one of these two questions, I think it is ok to add this annotation.

Comment (shokakucarrier) [reply src/main/java/org/commonjava/util/sidecar/interceptor/MetricsHandlerInterceptor.java:39]: It's must for any class without any constructor call, the annotation is only for compiler to not cut out these pieces of code, other reflected class don't need this.

**Pull request merged**